### DISCUSSION TITLE
Cant get "forced ota" working with local file

### DISCUSSION THREAD

**Discussion by git-greatguy** [Koenkk/zigbee2mqtt#13850] [Devices]
i'm trying to downgrade a zigbee device firmware throught local ota, but nothing work as expected.

here is my setup :

zigbee2mqtt as docker module for home assistant

zigbee2mqtt configuration.yaml for ota :

```
ota:
  zigbee_ota_override_index_location: my_index.json
```

content of /config/zigbee2mqtt/my_index.json :

```
[
    {
        "fileVersion": 5,
        "fileSize": 240478,
        "manufacturerCode": 4151,
        "imageType": 1,
        "sha512": "692417e8863827f120cda063850aee65b7b6eb58109d419ff58dced1d7c9251617312368bda0f8488aa26aeb7a8e3642339c7648af24cbff66b870401c9113da"      
        "url": "ZLinky_router_v5.0.ota",
        "force": true
    }
]
```
(i tried to keep only ota file reference and force=true and any difference)

then i tried ota check and here is the result :

```
Info 2022-09-04 11:35:09MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Checking if update available for 'Lixee_zlinky'","meta":{"device":"Lixee_zlinky","status":"checking_if_available"},"type":"ota_update"}'
Debug 2022-09-04 11:35:09Check if update available for '0x00158d0005ce7980' (ZLinky_TIC)
Debug 2022-09-04 11:35:09Using endpoint '1'
Debug 2022-09-04 11:35:11Received Zigbee message from 'Lixee_zlinky', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":6,"imageType":1,"manufacturerCode":4151}' from endpoint 1 with groupID 0
Debug 2022-09-04 11:35:11Got OTA request '{"fieldControl":0,"manufacturerCode":4151,"imageType":1,"fileVersion":6}'
Info 2022-09-04 11:35:11Using firmware file ZLinky_router_v6.0.ota
Info 2022-09-04 11:35:12No update available for 'Lixee_zlinky'
```

i cant see ant reference to my overrided ota file anywhere in logs. did i miss something ?

Do you have any clue of what's wrong with my setup ?

**Comment by Danit2** [comment]
I know your problem. 
I have the same. 

Read this: https://github.com/Koenkk/zigbee2mqtt/issues/12554

To solve this you habe to create a external converter form your device and then you have to change the 'ota.lixee' to 'ota.zigbeeOTA'

Then the local ota file work.

**Reply by git-greatguy** [reply]
thanks to point me this out. i'll go this way. i know nothing about ota procedure and converters, but with some reading ant some patience i'll learn someting knew and got it fashed :D

**Comment by git-greatguy** [comment]
ok, sounds really simple when we know where to look at. 
Correct me if i'm wrong but, converter for lixee already exist in /app/node_modules/zigbee-herdsman-converters/devices/lixee.js.
in file, there this this declaration :
line 644
`ota: ota.lixee,`

so, if i replace with  ota.zigbeeOTA, i'll be able to override ? that's it ? :D

**Comment by git-greatguy** [comment]
looks promising :)

```
2022-09-04 23:16:52ZigbeeOTA: Loading override index https://www.*****.eu/index.json
Debug 2022-09-04 23:16:52Is new image available for '0x00158d0005ce7980', current '{"fieldControl":0,"manufacturerCode":4151,"imageType":1,"fileVersion":6}', latest meta '{"url":"https://www.*****.eu/ZLinky_router_v5.0.ota","force":true}'
Info 2022-09-04 23:16:52MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"Lixee_zlinky","updateAvailable":true},"status":"ok","transaction":"s60w0-1"}'
```

thanks again !

**Comment by git-greatguy** [comment]
`MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"from":{"date_code":"20220830","software_build_id":"4000-0006"},"id":"Lixee_zlinky","to":{"date_code":"20220217","software_build_id":"4000-0005"}},"status":"ok","transaction":"0httf-1"}'`

:D
